larray/core/axis.py

```diff
@@ -1454,6 +1454,73 @@ def __dir__(self):
     def __iter__(self):
         return iter(self._list)
 
+    # TODO: move a few doctests to unit tests
+    def iter_labels(self, axes=None, ascending=True):
+        r"""Returns a view of the axes labels.
+
+        Parameters
+        ----------
+        axes : int, str or Axis or tuple of them, optional
+            Axis or axes along which to iterate and in which order. Defaults to None (all axes in the order they are
+            in the collection).
+        ascending : bool, optional
+            Whether or not to iterate the axes in ascending order (from start to end). Defaults to True.
+
+        Returns
+        -------
+        Sequence
+            An object you can iterate (loop) on and index by position.
+
+        Examples
+        --------
+
+        >>> from larray import ndtest
+        >>> axes = ndtest((2, 2)).axes
+        >>> axes
+        AxisCollection([
+            Axis(['a0', 'a1'], 'a'),
+            Axis(['b0', 'b1'], 'b')
+        ])
+        >>> axes.iter_labels()[0]
+        (a.i[0], b.i[0])
+        >>> for index in axes.iter_labels():
+        ...     print(index)
+        (a.i[0], b.i[0])
+        (a.i[0], b.i[1])
+        (a.i[1], b.i[0])
+        (a.i[1], b.i[1])
+        >>> axes.iter_labels(ascending=False)[0]
+        (a.i[1], b.i[1])
+        >>> for index in axes.iter_labels(ascending=False):
+        ...     print(index)
+        (a.i[1], b.i[1])
+        (a.i[1], b.i[0])
+        (a.i[0], b.i[1])
+        (a.i[0], b.i[0])
+        >>> axes.iter_labels(('b', 'a'))[0]
+        (b.i[0], a.i[0])
+        >>> for index in axes.iter_labels(('b', 'a')):
+        ...     print(index)
+        (b.i[0], a.i[0])
+        (b.i[0], a.i[1])
+        (b.i[1], a.i[0])
+        (b.i[1], a.i[1])
+        >>> axes.iter_labels('b')[0]
+        (b.i[0],)
+        >>> for index in axes.iter_labels('b'):
+        ...     print(index)
+        (b.i[0],)
+        (b.i[1],)
+        """
```

Comment:
I think examples above are very confusing for our users although they would be useful as a unittest. 
You should either print the labels or show how ``iter_labels`` can be used to iterate over data of an array.

Reply:
indeed.

Reply:
This is all about issue https://github.com/larray-project/larray/issues/447

In fact, since users should seldom use this directly (LArray.keys() and LArray.items() or zip_array_* should be what end users use) I am not very enthusiastic to change this (until 447 is done of course).

Reply:
OK